### PR TITLE
a gui!

### PR DESCRIPTION
**Pull request by nawnie**
this is a work in progress but and not tested. it will launch the gui and you can see what ive done, it should work since its just a launch code but ive got a little more left to do till im ready to test, anyone is free to finish this if they want

### CODE REVIEW

File: gui.py [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+        '--resume', str(resume),
+        '--Project_Name', project_name,
+        '--Learning_Rate', learning_rate,
+        '--Match_text_to_Unet', str(match_text_to_unet),
+        '--Text_lr', text_lr,
+        '--Schedule', schedule,
+        '--Text_lr_scheduler', text_lr_scheduler,
+        '--lr_warmup_steps', lr_warmup_steps,
+        '--lr_decay_steps', lr_decay_steps,
+        '--Text_lr_warmup_steps', text_lr_warmup_steps,
```
Comment (victorchall) [inline gui.py:90]: this arg doesn't exist

File: gui.py [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+        '--Text_lr_decay_steps', text_lr_decay_steps,
```
Comment (victorchall) [inline gui.py:91]: this arg doesn't exist

File: gui.py [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+open_file_button.grid(row=0, column=2)
+
+gradient_steps_label = tk.Label(tab2, text='Gradient Steps:')
+gradient_steps_label.grid(row=1, column=0, sticky=tk.W)
+
+gradient_steps_entry = tk.Entry(tab2)
+gradient_steps_entry.grid(row=1, column=1)
+gradient_steps_entry.insert(tk.END, '4')
+
+clip_grad_norm_var = tk.BooleanVar()
```
Comment (victorchall) [inline gui.py:276]: you define this as a bool but it needs to be a number when passed to trainer

File: gui.py [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+        '--lr_decay_steps', lr_decay_steps,
+        '--Text_lr_warmup_steps', text_lr_warmup_steps,
+        '--Text_lr_decay_steps', text_lr_decay_steps,
+        '--Resolution', resolution,
+        '--Batch_Size', batch_size,
+        '--Gradient_steps', gradient_steps,
+        '--Dataset_Location', dataset_location,
+        '--Max_Epochs', max_epochs,
+        '--Sample_File', sample_file,
+        '--clip_grad_norm', str(clip_grad_norm),
```
Comment (victorchall) [inline gui.py:98]: you defined this as a boolean for the UI, this should be a number

File: gui.py [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+        '--lr_warmup_steps', lr_warmup_steps,
+        '--lr_decay_steps', lr_decay_steps,
+        '--Text_lr_warmup_steps', text_lr_warmup_steps,
+        '--Text_lr_decay_steps', text_lr_decay_steps,
+        '--Resolution', resolution,
+        '--Batch_Size', batch_size,
+        '--Gradient_steps', gradient_steps,
+        '--Dataset_Location', dataset_location,
+        '--Max_Epochs', max_epochs,
+        '--Sample_File', sample_file,
```
Comment (victorchall) [inline gui.py:97]: this is not an arg, this should be `--sample_prompts`  I think??

File: gui.py [hunk 6]
```diff
@@ -0,0 +1,418 @@
...
+        '--logdir', logdir,
+        '--log_step', log_step,
+        '--lowvram', str(lowvram),
+        '--resume_ckpt', str(resume_ckpt),
+        '--run_name', run_name,
+        '--sample_steps', sample_steps,
+        '--save_ckpt_dir', save_ckpt_dir,
+        '--save_every_n_epochs', save_every_n_epochs,
+        '--save_ckpts_from_n_epochs', save_ckpts_from_n_epochs,
+        '--scale_lr', str(scale_lr),
```
Comment (victorchall) [inline gui.py:114]: this was long ago deprecated, should be removed

File: gui.py [hunk 7]
```diff
@@ -0,0 +1,418 @@
...
+    log_step = str(log_step_entry.get())
+    save_every_n_epochs = str(save_every_n_epochs_entry.get())
+    save_ckpts_from_n_epochs = str(save_ckpts_from_n_epochs_entry.get())
+    seed = str(seed_entry.get())
+
+    # Construct the command
+    command = [
+        'python',
+        'train.py',
+        '--model_type', model_type,
```
Comment (victorchall) [inline gui.py:79]: I don't think model_type is a thing

File: gui.py [hunk 8]
```diff
@@ -0,0 +1,418 @@
...
+        '--Save_optimizer_state', str(save_optimizer_state),
+        '--resume', str(resume),
+        '--Project_Name', project_name,
+        '--Learning_Rate', learning_rate,
+        '--Match_text_to_Unet', str(match_text_to_unet),
```
Comment (victorchall) [inline gui.py:84]: not an arg.

File: gui.py [hunk 9]
```diff
@@ -0,0 +1,418 @@
...
+        '--Text_lr', text_lr,
```
Comment (victorchall) [inline gui.py:85]: not an arg

File: gui.py [hunk 10]
```diff
@@ -0,0 +1,418 @@
...
+        '--Schedule', schedule,
+        '--Text_lr_scheduler', text_lr_scheduler,
```
Comment (victorchall) [inline gui.py:87]: not an arg

File: gui.py [hunk 11]
```diff
@@ -0,0 +1,418 @@
...
+        'python',
+        'train.py',
+        '--model_type', model_type,
+        '--Save_optimizer_state', str(save_optimizer_state),
+        '--resume', str(resume),
+        '--Project_Name', project_name,
+        '--Learning_Rate', learning_rate,
+        '--Match_text_to_Unet', str(match_text_to_unet),
+        '--Text_lr', text_lr,
+        '--Schedule', schedule,
```
Comment (victorchall) [inline gui.py:86]: not an arg, I think you meant lr_scheduler

**Comment by victorchall** [comment]
TLDR: I think you have a lot of bogus args that are supposed to be used to edit optimizer.json and such like you have in the colab, and not passed directly into the train.py as args.

**Comment by nawnie** [comment]
This should of been updated.

File: gui.py (hunk 11 above)
Comment (nawnie) [reply gui.py:86]: Ty for the info, this was the updated one. Gradient steps is for the gradient... I can't remember the arg but to increase virtual batch size batch size 4 x 2 gradient steps = 8 per step. 

I'll get the rest of this cleaned up shortly.  Still havnt gotten around to getting ed2 running on local Again.

Comment (victorchall) [reply gui.py:86]: that's grad_accum not gradient_steps.

**Comment by nawnie** [comment]
Ya in Colab labeled it gradient steps after explaining grad accumulation.
I'd post these easy fixes but havnt even had time to fix colab.
